packages/api/src/account/queries/delete_account_archive.sql

```diff
@@ -0,0 +1,6 @@
+DELETE FROM account_archive
+WHERE
+  archiveid = :archiveId
+  AND accountid IN (SELECT accountid FROM account WHERE primaryemail = :email)
```

Comment:
For some added defensiveness, I'd include a check that this `account_archive` isn't an ownership relationship.